protractor.sh

```diff
@@ -23,6 +23,11 @@ certutil -d "sql:$HOME/.pki/nssdb" -A -n console -t Pu,, -i /tmp/console-e2e.pem
 
 os::log::info "Launching protractor tests"
 
+# protractor will need this to login...
+service_account_token=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
+
+export SERVICE_ACCOUNT_TOKEN=$service_account_token
```

Comment:
Oh yeah!  So I originally also had:

```bash 
service_account_token=$(</var/run/secrets/kubernetes.io/serviceaccount/token)
if [[ -z "${service_account_token }" ]]; then
 echo "Failure to acquire service account token"
 exit 1
fi
export SERVICE_ACCOUNT_TOKEN=$service_account_token
```
but it was acting up so I cut the `if` statement for now.  Its probably a good idea to put it back in, though I suppose the default could be to run & do the normal login flow.  Thoughts?

Reply:
yeah lets put it back, just to be sure

Reply:
Also not really sure `var/run/secrets/kubernetes.io/serviceaccount/token` get into the container ? are we mounting it there ?

Reply:
From what I gather talking to @enj this is where tokens are mounted.  I built the image & deployed it via our `pod.yaml` and it did read out a token & `echo` to stdout.  Passing it into protractor & `console.log` also worked fine.

So just tinkering now with the `localStorage` bit to see if that will actually work.

Reply:
It is always there when SAs are enabled, which they always are in OS.

Reply:
For my own understanding, would that not be the case in vanilla `kubernetes` or another flavor?

Reply:
SAs are an optional (but highly recommended) feature in Kube.  Thus if that feature was disabled, none of the associated controllers would run, meaning there would be no token mounted in the pod (and even if there was a token, there would be no way to use that token for authentication purposes).

Reply:
Make sense.. for local development its ok to have it hardcoded or be passed as envVar in which case we could check if the `SERVICE_ACCOUNT_TOKEN` already exists.
If it does then we can assume that the container is started manually by `docker run ...` and the envVar is passed, if it's not present the get it from the mounted file.
Since we expect to tun the image in OS, I think we are(should be) safe. 
@benjaminapetersen thoughts ?

Reply:
Yeah, I'll factor that hard-coded one out.  Agree:
- if token exists via env var, skip login (running on cluster)
- else do login flow (running locally)